python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -330,6 +331,14 @@ def _report_current_usage(
             f"{cur_usage.overall.object_store_memory_str()}/"
             f"{limits.object_store_memory_str()} object_store_memory"
         )
+        if not cur_usage.overall.satisfies_limit(limits):
```

Comment:
since `satisfies_limit()` also checks cpu/gpu, we should directly compare the values of `cur_usage.overall.object_store_memory` and `limits.object_store_memory` (for example, we can first calculate `spilled_obj_memory`, then check if `spilled_obj_memory > 0`).

Reply:
Thanks for the quick suggestion!
Since object_store_memory could be `None`, I will also check if they are `None` or not.

Reply:
@scottjlee I have changed to only calculate difference of object_memory.

But I am not sure how to update with current master branch.
Should I use `git pull . upstream/master` in my local branch and push with `force-with-lease`?
Or should just click the "Update branch with merge commit" in this page?

Thanks again for your time.

Reply:
Hi, this looks good! The "Update branch" button works, but you shouldn't need to update with master if there are no conflicts. Could you take some screenshots so we can see this in action? There are examples that spill in `python/ray/data/tests/test_stats.py::test_spilled_stats`.  You can also add to this test by checking the bar's description . Thanks!

Reply:
Sure, I won't update with master if there are no conflicts.
Would you tell me more about what action I need to take some screenshots? Do you mean the changed `report_current_usage` ?

I will also check the tests in test_stats.py. Thanks!

Reply:
I have tried to add assertion to check the bar's description.
But I am struggling to getting the description from the executor from the dataset.
I have tried:
1. Mocking the ProgressBar, but cannot access the executor using `patch.object(...)`
2. Mocking the `tqdm_ray.tqdm` class, but I cannot access the instance attribute `_desc`.
3. Trying to access _current_executor from the dataset, but it remains as `None`.
4. Mocking `sys.stdout` with a new `StringIO` object, but it remains as an empty string.

It would be great if you can point me into the right direction in order to check the description.

Reply:
The progress bar gets closed after execution finishes, so I think you should be using `patch` instead of `patch.object`. You could try mocking both the progress bar or its `set_description` method?

Reply:
Thanks for the suggestions!
I used `patch` to mock the progress bar's `set_description` and check if there is any "spilled external storage" in the call args list.